website/pages/blog/building-cloudquery.md

```diff
@@ -0,0 +1,102 @@
+
+### Typing System
+
+The job of an ELT platform is to mostly do two things:
+
+1. Transform and normalize the data from an API
+2. For each transformed field add type data.
+
+CloudQuery transforms every single field to its own rich type system, which contains more than [17 types](https://github.com/cloudquery/plugin-sdk/tree/main/schema) (including things like IP Addresses, MAC Addresses) This ensures all the validation is happening in the transformation phase and then all destination plugins needs to do is just transform it depending on what types each destination is supporting. This is a big shift from how Singer or Airbyte works where they mostly use what is available in JSON and [JSON Schema](https://json-schema.org/) as they couple the encoding together with their type system.
+
+With those two decisions CloudQuery already supports more than [5 destination](https://www.cloudquery.io/docs/plugins/destinations) two months after our initial [V1 release](https://www.cloudquery.io/blog/cloudquery-v1-release)
+
+## Performance
+
+Cloud Infrastructure hyperscale created new challenges for the ELT world, For example some companies have more than 10K accounts GCP and/or AWS accounts with more than 100M resources. How do you fetch those on a daily (or twice a day)? To address this we made two architecture decisions:
+
+### Concurrency and Scheduling Model
+
+As part of our SDK which provides the concurrency model for all the source plugins we took advantage of the excellent support in Go for concurrency with Goroutines which enables us to scale to tens of thousands of API calls with low number of OS threads and low memory overhead.
+
+The second thing we did here is to create a good scheduler. Extracting APIs in concurrent matter involves restrictions on memory and this is similar to the classic computer science of crawling on how to efficiently crawl the web where you don’t know the number of links in each “depth”. CQ SDK is built in such a way that can support multiple scheduling algorithms and mechanisms with the default one being a concurrent DFS algorithm which makes sure memory usage is being kept at a given limit while taking advantage of all compute resources available for concurrency.
+
+### Horizontal Scaling
+
+The previously described in-process concurrency and scheduling algorithm gives a good utilization of compute and memory for one CQ process which makes it easy to scale vertically but what about horizontal scaling? To be able to scale both vertically and horizontally we designed CQ as stateless - i.e it doesn’t have any backend and it runs with just the right credentials and configuration file. This gives the ability for users to split configuration (for example configuration per account) and run CQ on as many nodes as possible which will work without any need for orchestration.
```

Comment:
Remove `in-process`, add comma, `as may nodes as possible` -> `as many nodes as needed`:

The previously described concurrency and scheduling algorithm gives a good utilization of compute and memory for one CQ process which makes it easy to scale vertically - but what about horizontal scaling? To be able to scale both vertically and horizontally we designed CQ as stateless - i.e. it doesn’t have any backend and it runs with just the right credentials and configuration file. This gives the ability for users to split configuration (for example configuration per account) and run CQ on as many nodes as needed which will work without any need for orchestration.